lib/resque/stat.rb

```diff
@@ -7,12 +7,19 @@ module Resque
   #   Kill a stat: Stat.clear(name)
   module Stat
     extend self
-    
-    # Direct access to the Redis instance.
+
     def redis
-      Resque.redis
+      warn '[Resque] [Deprecation] Resque::Stat #redis method is deprecated (please use #data_strore)'
+      data_store
+    end
+
+    def data_store
+      @data_store ||= Resque.redis
```

Comment:
This caching is a bit useless (no performance gain) and seems even dangerous or at least counterintuitive. If I change `Resque.redis` to point to somewhere else, I would expect things to work correctly, but since this module here now has the "old" value memoized, it would continue using the old one.